### DISCUSSION TITLE
Categorie for QBIT is necessary

### DISCUSSION THREAD

**Discussion by TheNexter** [jesec/flood#270] [Ideas]
https://github.com/jesec/flood/issues/81

Hello man, yes I know you already answered this subject but I think this is really the most important function of all that could exist, let me explain:

qbittorrent is complete software with proxy, management files perfects, work with radarr and sonarr without issue and all feature work.

Transmission is to light, no proxy, no good files managements, no native interface correct to settings somethings, not a realy good downloads software.

rutorrent is not compatible with proxy, no good interface for files management and allot of other things.

If you work and release categorie, it's can be a game changer because allot of millions people use radarr / sonarr, and we need to everytimes go on ugly interface of qbit.

Only one thing can do a big and huge difference

https://hotio.dev/containers/qflood/

The team Hotio have understand that and they have a docker image just for your software 😎

Thx for your HUGE work

And sorry for my VERY bad english 😥

**Comment by jesec** [comment]
Moved to discussions as this is a feature request.

This will not be implemented. Tags render categories obsolete.

**Reply by agneevX** [reply]
Hi! in qBittorrent, tags are used only for organizing only. 

Individual save paths can be specified in each category so it's very helpful if you have multiple storage locations. 

Personally, I use categories for every single new addition.

**Reply by jef** [reply]
> Personally, I use categories for every single new addition.

I believe this is very common among qB users. I myself do this and don't use tags. I believe tags are essentially for folks that also "manage" their content within qB rather than having Sonarr/Radarr or other providers do this.

qBittorrent: Category == rTorrent: Label; Transmission has no such capability with it's daemon (I believe).

**Reply by jef** [reply]
Currently working on adding the filter and will test do some tests later.

**Reply by jesec** [reply]
Unfortunately I am not going to accept such addition. IMO category is completely unnecessary. qBittorrent is also weighing to basically ditch category in PR 13773. 

Plus, it is not going to work with other torrent clients.

**Reply by agneevX** [reply]
FWIW, I've never been actually able to add torrents via Flood because of this.

**Reply by jef** [reply]
https://github.com/qbittorrent/qBittorrent/pull/13773 is not ditching categories, it's removing the automatic save path management, which is related to categories. For those who are unfamiliar with qbt, the danger here is that if you have N torrents categorized as Y and you change Y's save path from A to B, then all torrents under Y would be moved to B. N can be quite large (> 10000) and moving that many files by a mistake is the danger they are trying to circumvent here.

---

As for your opinion, what else do you see for an option to manage a torrent's location within qBittorrent? rTorrent has been using labels (categories per qbt) for a long time and transmission doesn't have them. Thus, GUIs like transgui has implemented their own labels (categories per qbt) (https://github.com/transmission-remote-gui/transgui/pull/1305).

qBittorrent may be the only one with this verbiage, but is similar to both tools. It's the only one with both categories and labels. IMO labels are irrelevant in many use cases for qbt, but they have them. They don't provide anything except filtering in the UI. As for categories, it's great for media management. I believe you understand this, but I'm trying to figure out (in your opinion) why this filtering doesn't matter in flood when you support label filter for rTorrent.

**Reply by jesec** [reply]
Transmission does support tags/labels. [1]

***

To manage location, just type in the destination when you add torrents. Categories are not useful. Tags can do it all. It is possible to assign multiple tags to one torrent but only one category can be assigned to a torrent. Category has no additional significance in Flood, and it will not differ from the tag, except that it is inferior. Plus, Flood suggests destination based on tags only. 

Note that I assume that Flood is the one and only web UI for that torrent client. That means in no circumstance should a feature depends on other web UIs. Unfortunately category is not a good fit here, as the complexity would be too great and unnecessary to implement it. 

[1]: https://github.com/transmission/transmission/blob/8566df069899ce8923463cadeb0ff66d4544991a/extras/rpc-spec.txt#L120

**Reply by agneevX** [reply]
> To manage location, just type in the destination when you add torrents.

That's not possible for every single torrent, when stored in multiple folders across drive pools. 

Categories save complex file locations into a single option, that's why it's used. 

> Categories are not useful. Tags can do it all.

Sorry but you're completely wrong. Almost all applications that integrate with qBittorrent use categories because of an exclusive save path (even if not specified).

> It is possible to assign multiple tags to one torrent but only one category can be assigned to a torrent.

That's because tags are exclusively used for organization and nothing else. It's purely cosmetic and is ideal for folks who have say, `movies`, `music` and `images`. 
Those are 3 tags. If you need this sort of micro-management, then tags are useful but again, is not a replacement for categories under any circumstance.

**Reply by zotabee** [reply]
Flood is awesome but would be really much better if it supported categories. This is really much useful when having a lot of data and torrents.

> Unfortunately I am not going to accept such addition. IMO category is completely unnecessary. qBittorrent is also weighing to basically ditch category in PR 13773.

I don't see where they stated that they wanted to remove categories from qBittorrent. Maybe I misunderstood or this has been discussed somewhere else than https://github.com/qbittorrent/qBittorrent/pull/13773

> Plus, it is not going to work with other torrent clients.

Unfortunately, rTorrent and Deluge are basically dead projects since a long time now so if it's to wait of these projects for features that aren't implemented and that will never come. (Telling that as a rTorrent/ruTorrent user for the last 10 years and finally making the switch these days to qBittorrent and Flood). Transmission is mainly macOS focused or let say in the spirit at least.

qBittorrent is the last great BitTorrent client still alive and updated daily which is cross platform and supported on servers and desktops platforms. Would be good to focus perhaps more on this client even if other clients don't have all the features.

I really appreciate your awesome work @jesec but I don't understand the justification here.

**Reply by glolsh** [reply]
I'm with others. Would be great if you add categories in future to this awesome UI, that you're developing.

I'm supporting hundreds of torrents (somebody supports thousands) and categories really help out here. Typing out path everytime is painful

**Comment by raphaelfuchter** [comment]
Sonarr and Radarr use the categories to know what should be imported and what is completed. This is an essential feature for those who use these 2 services.

**Reply by jesec** [reply]
Nope. *arrs supports Flood natively, and they work just fine without categories.

**Reply by raphaelfuchter** [reply]
How do you know what the *arrs have already completed the import? I use as categories "movie" and "movie-imported", so that I know what is already completed / imported.

**Reply by jesec** [reply]
![image](https://user-images.githubusercontent.com/11585141/119829419-cbe2bd00-bf2d-11eb-83ac-408e7440c5be.png)

**Reply by Nindouja** [reply]
@jesec In the use case when I manually added a .torrent file to flood (of an existing film/show in radarr/sonarr). Is there any way to make *arr import it on download finish without categories ?

This is a common use case I believe for trackers that don't support automation. I always have to manually add a torrent in my language into qbittorrent and when I do that they are automatically handled in *arr upon completion **if I set the correct categories on them**.

Am I missing something ? Is that doable without categories ? Anyway thanks for the good work; flood look amazing !

**Comment by C84186** [comment]
I understand @jesec 's reluctance to use categories, however, I'm personally not clear on what processes to use to make tags behave similarly.

To list the differences:

- Categories are mutually exclusive, tags are not
- In qBT, categories typically dictate some logic about save paths

Personally, I like the save path stuff well enough, but IMO it's not robust enough to make categories *that* useful, for example:

if you wanted to download all seasons of a show, but torrents were only available for seasons individually, in order to make a media server treat the show correctly, we must get the show into `Category_TV/Show/Season/Episode` structure. 

Categories do not adequately address this.

This being said, I'm still not satisfied there's an equivalent in flood

I'm aware that @jesec believes that `Flood` should be where users set up their paths, without making their clients do any magic behind the scenes, but presently flood doesn't provide a lot of utility to help with this, and I'm left confused what my backend client should be doing vs what flood should be doing.

**Comment by C84186** [comment]
As I mentioned in #181 , I've implemented a workaround for this, that basically uses tags with special prefixes to get / set category state of a torrent. 
https://github.com/C84186/qBittorrent_tag_by_pattern/blob/main/synchronize_cat_tags.py

Say you have a category, "Movies", and a torrent "Tangerine (2015)"

To categorize that torrent as a movie, tag it with "_SET_Movies". 
When the script runs, it's category is assigned, the "_SET" tag is removed, and you will see it now tagged as "_CAT_Movies". 

If you cron the script to run every 10 minutes, it should be enough to have a degree of control over categories.

**Comment by benfishbus** [comment]
Flood UI is really nice. I would love to use it with qBittorrent, but categories are essential. Transmission and rtorrent/rutorrent have their ways of facilitating torrent data relocation; qB has categories. Browser extensions, *arr apps, etc. all leverage categories when working with qB. The feature has been around a long time, and doesn't look like it's going anywhere.

**Comment by scratchyone** [comment]
i want to add to this and say that categories are essential for me as well. tags are useful, but only categories allow you to manage save paths. for example, i use categories to control which items are synced with my local plex server.

if i want to grab an item from my seedbox, i place it into the "sync" category which causes it to be automatically moved by qbit into a folder that syncs automatically to my local server. flood isn't really usable with qbit without this feature
